aws/resource_aws_efs_access_point_test.go

```diff
@@ -0,0 +1,515 @@
+		CheckDestroy: testAccCheckEfsAccessPointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSAccessPointConfigPosixUserSecondaryGids(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "posix_user.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "posix_user.0.gid", "1001"),
+					resource.TestCheckResourceAttr(resourceName, "posix_user.0.uid", "1001"),
+					resource.TestCheckResourceAttr(resourceName, "posix_user.0.secondary_gids.#", "1")),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSEFSAccessPoint_tags(t *testing.T) {
+	var ap efs.AccessPointDescription
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_efs_access_point.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckEfsAccessPointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSAccessPointConfigTags1(rName, "key1", "value1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEFSAccessPointConfigTags2(rName, "key1", "value1updated", "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+			{
+				Config: testAccAWSEFSAccessPointConfigTags1(rName, "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSEFSAccessPoint_disappears(t *testing.T) {
+	var ap efs.AccessPointDescription
+	resourceName := "aws_efs_access_point.test"
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVpnGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSAccessPointConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckEfsAccessPointExists(resourceName, &ap),
+					testAccAWSEFSAccessPointDisappears(&ap),
```

Comment:
This can be simplified to the new `testAccCheckResourceDisappears()` helper. See also the new [Contributing Guide documentation](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#disappears-acceptance-tests) 👍 e.g.

```go
					testAccCheckResourceDisappears(testAccProvider, resourceAwsEfsAccessPoint(), resourceName),
```